File: python-beginner-workbook/module-06-oop/examples/example-03-class-with-method.py
# example-03-class-with-method.py
# Demonstrates class with methods

class Customer:
    def __init__(self, name, email):
        self.name = name
        self.email = email
    
    def display(self):
        print(f"Customer: {self.name}")
        print(f"Email: {self.email}")
    
    def update_email(self, new_email):
        self.email = new_email

# Create customer
customer = Customer("Sarah", "sarah@example.com")
customer.display()

customer.update_email("sarah.new@example.com")
customer.display()

File: csharp-beginner-workbook/module-06-oop-intro/examples/example-03-class-with-method.cs
// Example 3: Class with Method
// This demonstrates adding methods to classes so objects can perform actions

// Define the Customer class with methods
class Customer
{
    public string Name { get; set; }
    public string Email { get; set; }
    public bool IsVip { get; set; }
    
    // Method that displays customer information
    public void DisplayInfo()
    {
        string vipStatus = IsVip ? "VIP Member" : "Regular Customer";
        Console.WriteLine("=== Customer Profile ===");
        Console.WriteLine($"Name: {Name}");
        Console.WriteLine($"Email: {Email}");
        Console.WriteLine($"Status: {vipStatus}");
        Console.WriteLine();
    }
    
    // Method that upgrades customer to VIP
    public void UpgradeToVip()
    {
        IsVip = true;
        Console.WriteLine($"{Name} has been upgraded to VIP status!");
    }
}

// Main program
Customer customer1 = new Customer();
customer1.Name = "Sarah Johnson";
customer1.Email = "sarah@example.com";
customer1.IsVip = false;

Customer customer2 = new Customer();
customer2.Name = "Mike Chen";
customer2.Email = "mike@example.com";
customer2.IsVip = true;

// Use the methods
Console.WriteLine("=== Customer Database ===\n");
customer1.DisplayInfo();
customer2.DisplayInfo();

// Upgrade customer1
Console.WriteLine("=== Upgrading Customer ===");
customer1.UpgradeToVip();
customer1.DisplayInfo();

